packages/tpetra/core/src/Tpetra_Import_Util2.hpp

```diff
@@ -1237,28 +1250,263 @@ lowCommunicationMakeColMapAndReindexKokkos (const Teuchos::ArrayView<const size_
   Kokkos::deep_copy(execution_space(), colind_LID_host, colind_LID_view);        
 }
 
-
 template <typename LocalOrdinal, typename GlobalOrdinal, typename Node>
-void lowCommunicationMakeColMapAndReindex (const Teuchos::ArrayView<const size_t> &rowptr,
-                                      const Teuchos::ArrayView<LocalOrdinal> &colind_LID,
-                                      const Teuchos::ArrayView<GlobalOrdinal> &colind_GID,
+void
+lowCommunicationMakeColMapAndReindex (
+                                      const Kokkos::View<size_t*,typename Node::device_type>        rowptr_view,
+                                      const Kokkos::View<LocalOrdinal*,typename Node::device_type>  colind_LID_view,
+                                      const Kokkos::View<GlobalOrdinal*,typename Node::device_type> colind_GID_view,
                                       const Teuchos::RCP<const Tpetra::Map<LocalOrdinal,GlobalOrdinal,Node> >& domainMapRCP,
                                       const Teuchos::ArrayView<const int> &owningPIDs,
                                       Teuchos::Array<int> &remotePIDs,
                                       Teuchos::RCP<const Tpetra::Map<LocalOrdinal,GlobalOrdinal,Node> > & colMap)
 {
-  // Running lowCommMakeColMap with Kokkos is slower on host, but faster on device
-  // If Kokkos runs on host, select the serial, non-Kokkos version of lowCommMakeColMap, 
-  // but if Kokkos runs on device (CUDA / HIP), select the Kokkos version of lowCommMakeColMap
+  using Teuchos::rcp;
+  typedef LocalOrdinal LO;
+  typedef GlobalOrdinal GO;
+  typedef Tpetra::global_size_t GST;
+  typedef Tpetra::Map<LO, GO, Node> map_type;
+  const char prefix[] = "lowCommunicationMakeColMapAndReindex: ";
+
   typedef typename Node::device_type DT;
-  static constexpr bool runOnHost = !std::is_same_v<typename DT::execution_space, Kokkos::DefaultExecutionSpace> || std::is_same_v<Kokkos::DefaultExecutionSpace, 
-    Kokkos::DefaultHostExecutionSpace>;
-  if(runOnHost) {
-    lowCommunicationMakeColMapAndReindexSerial<LocalOrdinal, GlobalOrdinal, Node> (rowptr, colind_LID, colind_GID, domainMapRCP, owningPIDs, remotePIDs, colMap);
+  using execution_space = typename DT::execution_space;
+  execution_space exec;
```

Comment:
It would be nice to use ``exec`` consistently throughout the function (instead of ``execution_space()`` and ``DT()``, in case we use a non-default stream later on.

Reply:
I'm not sure how I'd use `exec` instead of `DT()`, can you explain?

Reply:
For example, below on line 1279 you can replace `DT()` with `exec` and it will work just fine, same thing on line 1465.
Although you might want to first call the constructor on `exec`.

Reply:
done